app/src/main/java/com/xinto/opencord/ui/viewmodel/ChannelPinsViewModel.kt

```diff
@@ -6,14 +6,15 @@ import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
 import androidx.lifecycle.viewModelScope
 import com.xinto.opencord.domain.manager.PersistentDataManager
+import com.xinto.opencord.domain.mapper.toDomain
 import com.xinto.opencord.domain.model.DomainMessage
-import com.xinto.opencord.domain.repository.DiscordApiRepository
+import com.xinto.opencord.rest.service.DiscordApiService
 import com.xinto.opencord.ui.viewmodel.base.BasePersistenceViewModel
 import kotlinx.coroutines.launch
 
 class ChannelPinsViewModel(
     persistentDataManager: PersistentDataManager,
-    private val repository: DiscordApiRepository
+    private val api: DiscordApiService
```

Comment:
Why pass service to the viewmodel? shouldn't we pass the store and call the service inside the store class?

Reply:
oops

Reply:
resolve this and then mergussy